compiler/grainc/dune

```diff
@@ -32,5 +26,4 @@
  (modes exe js)
  (libraries grain grain_diagnostics binaryen.js dune-build-info js_of_ocaml)
```

Comment:
Do you still need the js_of_ocaml dependency ?

Reply:
I think I can remove that now! Thanks for pointing it out